### PR TITLE
ADE SP KV URL equivalence

### PR DESCRIPTION
**Pull request by ejarvi**
This change considers KV URL's that only differ in case or trailing slash to be the same
https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name

### CODE REVIEW

File: VMEncryption/main/ExtensionParameter.py [hunk 1]
```diff
@@ -156,6 +156,13 @@ def clear_config(self):
     def _is_encrypt_command(self, command):
         return command in [CommonVariables.EnableEncryption, CommonVariables.EnableEncryptionFormat, CommonVariables.EnableEncryptionFormatAll]
 
+    def _is_kv_equivalent(self, a, b):
+        # ignore trailing slash if present, and ensure case insensitive string comparison for key vault and object name
+        # https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name 
+        if a and a[-1] == '/': a = a.lower()[:-1]
```
Comment (vimish) [inline VMEncryption/main/ExtensionParameter.py:162]: Can the python's rstrip function be used to remove / from the end of the string?
Thinking of case if customers by multiple / at the end of the url.

**Comment by ejarvi** [comment]
I hadn't considered that to be a valid KeyVault URL (the key vault spec doesn't allow for multiple trailing slashes for URI's).  I also have not observed that in any telemetry like we have the other cases. Maybe adding some code to reject that case in check_util would be preferable.  

Right now the code for checkutil won't block this, urlparse just treats trailing slashes as part of the path: 
>>> from urllib.parse import urlparse
>>> urlparse("https://keyvault.windows.net/foo/foo/////")
ParseResult(scheme='https', netloc='keyvault.windows.net', path='/foo/foo/////', params='', query='', fragment='')

File: VMEncryption/main/ExtensionParameter.py (hunk 1 above)
Comment (ejarvi) [reply VMEncryption/main/ExtensionParameter.py:162]: I hadn't considered that to be a valid KeyVault URL (the key vault spec doesn't allow for multiple trailing slashes for URI's). I also have not observed that in any telemetry like we have the other cases. Maybe adding some code to reject that case in check_util would be preferable.

Right now the code for check_util won't block it though, urlparse just treats trailing slashes as part of the path:

from urllib.parse import urlparse
urlparse("https://keyvault.windows.net/foo/foo/////")
ParseResult(scheme='https', netloc='keyvault.windows.net', path='/foo/foo/////', params='', query='', fragment=''

**Review by vimish — APPROVED** [review]

File: VMEncryption/main/ExtensionParameter.py (hunk 1 above)
Comment (ejarvi) [reply VMEncryption/main/ExtensionParameter.py:162]: This PR that was not yet merged may be an option for blocking multiple trailing slashes or other URI's that do not meet the more restrictive KV rules:  https://github.com/ejarvi/azure-linux-extensions/pull/3/files

Comment (ejarvi) [reply VMEncryption/main/ExtensionParameter.py]: latest commit should cover this

**Pull request merged**